test/extended/openstack/egressip.go

```diff
@@ -227,6 +219,108 @@ spec:
 		e2e.Logf("Found the expected Fixed IP (%s) for the FIP '%s'", egressIPAddrStr, fip.FloatingIP)
 
 	})
+
+	// https://issues.redhat.com/browse/OCPBUGS-27222
+	g.It("with IPv6 format should be created on dualstack cluster with OVN-Kubernetes NetworkType and dhcpv6-stateful mode", func() {
+
+		networks, err := oc.AdminConfigClient().ConfigV1().Networks().Get(ctx, "cluster", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dualstack, err := isDualStackCluster(networks.Status.ClusterNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if !dualstack {
+			e2eskipper.Skipf("Test only applicable for dualstack clusters")
+		}
+
+		g.By("Getting the network type")
+		networkType, err := getNetworkType(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if networkType != NetworkTypeOVNKubernetes {
+			e2eskipper.Skipf("Test not applicable for '%s' NetworkType (only valid for '%s')", networkType, NetworkTypeOVNKubernetes)
+		}
+
+		e2e.Logf("Getting the worker for the EgressIP")
+		worker := workerNodeList.Items[0]
+
+		g.By(fmt.Sprintf("Getting the EgressIP network from the '%s' annotation", egressIPConfigAnnotationKey))
+		egressIPNetCidrStr, egressPortId, err := getEgressNetworkInfo(worker, "ipv6")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(egressIPNetCidrStr).NotTo(o.BeEmpty(), "Could not get the EgressIP network from the '%s' annotation", egressIPConfigAnnotationKey)
+		e2e.Logf("Found the EgressIP network: %s", egressIPNetCidrStr)
+		o.Expect(egressPortId).NotTo(o.BeEmpty(), "Could not get the Egress openstack portId from the '%s' annotation", egressPortId)
+		e2e.Logf("Found the Egress PortID: %s", egressPortId)
+
+		ipv6mode, err := getipv6ModeFromSubnetCidr(networkClient, egressIPNetCidrStr)
```

Comment:
Thanks to your comment I noticed something that is (or will be) problematic for upstream CI: There are several subnets in your project using the same CIDR. Therefore, I need something more specific to identify the subnet. We were using the infraID tag, but that's not applicable for dualstack because we are using BYON so we are creating the machine subnet.

I moved the logic to filter based on the networkID that is detected thanks to the portID that is written in the egressIP annotation in the worker :) That is currently working for ipv4, I am refactoring that part for ipv6 too.